controller/src/main/java/io/pravega/controller/store/stream/InMemoryStream.java

```diff
@@ -116,6 +116,9 @@
     @GuardedBy("subscribersLock")
     private final List<VersionedMetadata<StreamSubscriber>> streamSubscribers = new ArrayList<>();
 
+    @GuardedBy("subscribersLock")
+    private final Map<String, Long> subscribersSet = new HashMap<String, Long>();
```

Comment:
nit: misnomer `set` when its a `map`

Reply:
also why do we need an additional map here? can we update the StreamSubscriber to have the generation field?

Reply:
SubscriberSet was not created for adding generations. 
It exists so we can maintain a separate list of subscribers from the "Subscriber--> StreamCut" mapping. 
This this needed for finding the list of all Subscribers for a Stream (listSubscribers)

Reply:
> nit: misnomer `set` when its a `map`

changed to Subscribers